pkgs/development/web/nodejs/v20.nix

```diff
@@ -19,5 +19,15 @@ buildNodejs {
       url = "https://github.com/nodejs/node/commit/534c122de166cb6464b489f3e6a9a544ceb1c913.patch";
       hash = "sha256-4q4LFsq4yU1xRwNsM1sJoNVphJCnxaVe2IyL6AeHJ/I=";
     })
+    (fetchpatch2 {
+      # Starting from OpenSSL 3.0.14, 3.1.6, 3.2.2, and 3.3.1, OpenSSL was fixed
+      # to return an error reason string for bad/unknown application protocols.
+      # 
+      # This patch updates tests to handle both the old `ECONNRESET` error on older versions
+      # of OpenSSL and the new `ERR_SSL_TLSV1_ALERT_NO_APPLICATION_PROTOCOL` on
+      # newer versions of OpenSSL.
```

Comment:
```suggestion
      # Starting from OpenSSL 3.0.14, 3.1.6, 3.2.2, and 3.3.1, OpenSSL was fixed
      # to return an error reason string for bad/unknown application protocols.
      #
      # This patch updates tests to handle both the old `ECONNRESET` error on older versions
      # of OpenSSL and the new `ERR_SSL_TLSV1_ALERT_NO_APPLICATION_PROTOCOL` on
      # newer versions of OpenSSL.
      # It can be removed once it's backported upstream.
```